hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/test/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapterTest.java

```diff
@@ -225,9 +233,42 @@ public void testIncompatibleDeletesAreDropped() {
     Assert.assertEquals(
         Arrays.asList(put), incompatibleMutationAdapter.adaptIncompatibleMutations(walEntry));
 
-    verify(metricsExporter).incCounters(INCOMPATIBLE_MUTATION_METRIC_KEY, 0);
+    testInitialsMetrics();
     verify(metricsExporter, times(1)).incCounters(INCOMPATIBLE_MUTATION_METRIC_KEY, 1);
+    verify(metricsExporter, times(1)).incCounters(INCOMPATIBLE_MUTATION_DELETES_METRICS_KEY, 1);
     verify(metricsExporter, times(1)).incCounters(DROPPED_INCOMPATIBLE_MUTATION_METRIC_KEY, 1);
+    verifyNoMoreInteractions(metricsExporter);
+  }
+
+  @Test
+  public void testTimestampsOverflowMutations() {
+    ArrayList<Cell> walEntryCells = new ArrayList<>();
+    Cell put1 = new KeyValue(rowKey, cf, qual, Long.MAX_VALUE - 1, KeyValue.Type.Put, val);
+    Cell put2 = new KeyValue(rowKey, cf, qual, 0, KeyValue.Type.Put, val);
+    // make sure this is not flagged as overflow.
+    Cell delete =
+        new KeyValue(rowKey, cf, qual, HConstants.LATEST_TIMESTAMP, Type.DeleteFamily, val);
+    walEntryCells.add(put1);
+    walEntryCells.add(put2);
+    walEntryCells.add(delete);
+    BigtableWALEntry walEntry = new BigtableWALEntry(100L, walEntryCells, tableName);
+
+    Assert.assertEquals(
+        Arrays.asList(put1, put2, delete),
+        incompatibleMutationAdapter.adaptIncompatibleMutations(walEntry));
+
+    testInitialsMetrics();
+    verify(metricsExporter, times(1))
+        .incCounters(INCOMPATIBLE_MUTATION_TIMESTAMP_OVERFLOW_METRIC_KEY, 1);
+    verify(metricsExporter, times(1)).incCounters(INCOMPATIBLE_MUTATION_METRIC_KEY, 1);
+    verifyNoMoreInteractions(metricsExporter);
+  }
+
+  private void testInitialsMetrics() {
```

Comment:
I don't think we need this method. We can make the tests more DAMP [go/tott/598](https://goto.google.com/tott/598)

Reply:
Done.